src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/full_screen_waterfall/index.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { EmbeddableRenderer } from '@kbn/embeddable-plugin/public';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFocusTrap,
+  EuiOverlayMask,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { SERVICE_NAME_FIELD, SPAN_ID_FIELD, TRANSACTION_ID_FIELD } from '@kbn/discover-utils';
+import { i18n } from '@kbn/i18n';
+import { DocViewRenderProps } from '@kbn/unified-doc-viewer/types';
+import { useRootTransactionContext } from '../../doc_viewer_transaction_overview/hooks/use_root_transaction';
+import { SpanFlyout } from './span_flyout';
+
+export interface FullScreenWaterfallProps {
+  traceId: string;
+  rangeFrom: string;
+  rangeTo: string;
+  dataView: DocViewRenderProps['dataView'];
+  tracesIndexPattern: string;
+  onCloseFullScreen: () => void;
+}
+
+export const FullScreenWaterfall = ({
```

Comment:
Does it make sense to move this component to a generic place and it could accept a children property, which in this case it would be `EmbeddableRenderer`?

I see other places where we could reuse it like, the Trace Waterfall on APM, the Profiling flamegraph...